.github/workflows/auto-merge.yml

```diff
@@ -0,0 +1,22 @@
+name: Dependabot auto-merge
+on: pull_request_target
+
+permissions:
+  pull-requests: write
+  contents: write
+
+jobs:
+  dependabot:
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'dependabot[bot]' }}
```

Comment:
Strictly speaking, if-clauses do not need interpolation syntax and should only contain expressions.

Reply:
Tell that to Microsoft. :P
That's a verbatim copy/paste from the docs.
https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/automating-dependabot-with-github-actions#enable-auto-merge-on-a-pull-request